Civi/Share/Message.php

```diff
@@ -135,27 +138,96 @@ public function send($local_node_id = null)
    *
    * @return void
    */
-  public function processOnNode($local_node_id)
+  public function processChanges($local_node_id)
   {
     $lock = \Civi::lockManager()->acquire('data.civishare.changes'); // is 'data' the right type?
 
     // load the changes
     $changes = civicrm_api4('ShareChange', 'get', [
       'where' => [
         ['id', 'IN', $this->change_ids],
+        ['status', 'IN', [ShareChange::STATUS_PENDING]],
       ],
       'checkPermissions' => TRUE,
     ]);
 
-    // process them
+    // hand them over to the ChangeProcessor system
     foreach ($changes as $change) {
-      $this->applyChange($change);
+      $change_processor = new \Civi\Share\ChangeProcessingEvent($change['id'], $local_node_id, $change);
+      try {
+        \Civi::dispatcher()->dispatch('de.systopia.change.process', $change_processor);
```

Comment:
Let's define the event name in a constant on the event class and rename it to match the CiviCRM event naming schema, e. g. `civi.share.change.process`.